zwave_js_server/util/command_class.py

```diff
@@ -0,0 +1,49 @@
+"""Command Class specific utility functions for values."""
+from typing import Union
+
+from ..const import (
+    CC_SPECIFIC_METER_TYPE,
+    CC_SPECIFIC_SCALE,
+    CC_SPECIFIC_SENSOR_TYPE,
+    METER_TYPE_TO_SCALE_ENUM_MAP,
+    CommandClass,
+    MeterScaleType,
+    MeterType,
+    MultilevelSensorType,
+)
+from ..exceptions import InvalidCommandClass
+from ..model.value import Value
+
+
+def get_meter_type(value: Value) -> Union[MeterType, None]:
+    """Get the MeterType for a given value."""
+    if value.command_class != CommandClass.METER:
+        raise InvalidCommandClass(value, CommandClass.METER)
+
+    try:
+        return MeterType(value.metadata.cc_specific[CC_SPECIFIC_METER_TYPE])
+    except ValueError:
+        return None
```

Comment:
When could this happen? If the device has incorrect config?

Reply:
It's probably more likely that a new type gets introduced that we don't have in the enum yet. I came across an issue when testing this in the HA PR, but maybe we should start with just letting it fail and if we need to add guards we can. Things failing will otherwise be a forcing function for us to know that something is wrong

Reply:
Do you mean we should remove the try... except here?

Reply:
Oh I know what it was. It was causing a test to fail because we had an old fixture that did not have this `ccSpecific` data yet. It probably just makes more sense to fix the fixture then

Reply:
Looked into it again, and it was because of a bad interview. This was the value that was throwing the error:
```
{
            "endpoint": 3,
            "commandClass": 49,
            "commandClassName": "Multilevel Sensor",
            "property": "UNKNOWN (0x00)",
            "propertyName": "UNKNOWN (0x00)",
            "ccVersion": 0,
            "metadata": {
                "type": "number",
                "readable": true,
                "writeable": false,
                "label": "UNKNOWN (0x00)",
                "ccSpecific": {
                    "sensorType": 0,
                    "scale": 0
                }
            },
            "value": 0
        },
```
Do you still think we should avoid the guard? I can fix the `ccSpecific` values here but it won't prevent the exception from raising for users that have this issue

Reply:
If we are sure that the reason is a bad interview or similar I think we should raise a specific error letting the user know that something is wrong with the config of the device.

Reply:
OK, we will have to catch that in discovery then so we can continue with discovery of other values